src/receiver.ts

```diff
@@ -113,33 +114,43 @@ export default class Receiver extends EventEmitter {
                     try {
                         const response: Response = await this.fetch(item.url)
 
-                        // parse the notification
+                        const responseText = await response.text()
                     
-                        const jsonldParser = JsonLdParser.fromHttpResponse(
-                            response.url,
-                            response.headers.get('content-type') || "application/ld+json"
-                        )
-
-                        // transform bodystream
-                        //const bodyStream = new ReadableWebToNodeStream(response.body || new ReadableStream())
-
-                        // TODO: Fix this when NodeJS vs. Stream API chaos is over
-                        const bodyStream = new Readable()
-                        bodyStream.push(await response.text())
-                        bodyStream.push(null)
-
-                        // parse the notification
-                        const notification = await EventNotification.parse(bodyStream, jsonldParser)
-
-                        // emit an event with notification
-                        const idToCheck = strategy == 'notification_id' ? item.url : notification.id.value
-
-                        if (this.db && !this.db.get(idToCheck)) {
-                            this.emit('notification', notification)
-                            this.db.set(idToCheck, true)
+                        try {
+                            // parse the notification
+                            const jsonldParser = JsonLdParser.fromHttpResponse(
+                                response.url,
+                                response.headers.get('content-type') || "application/ld+json"
+                            )
+
+                            // transform bodystream
+                            //const bodyStream = new ReadableWebToNodeStream(response.body || new ReadableStream())
+
+                            // TODO: Fix this when NodeJS vs. Stream API chaos is over
+                            const bodyStream = new Readable()
+                            bodyStream.push(responseText)
+                            bodyStream.push(null)
+
+                            // parse the notification
+                            const notification = await EventNotification.parse(bodyStream, jsonldParser)
+
+                            // emit an event with notification
+                            const idToCheck = strategy == 'notification_id' ? item.url : notification.id.value
+
+                            if (this.db && !this.db.get(idToCheck)) {
+                                this.emit('notification', notification)
+                                this.db.set(idToCheck, true)
+                            }
+                        }
+                        catch (e) {
+                            const idToCheck = md5.default(responseText)
+                            if (this.db && !this.db.get(idToCheck)) {
```

Comment:
Why do unparseable notifications need to be added to the cache? Aren't they simply ignored?

Reply:
But if you get a lot of those, your cache blows up.
How about we split the error event in `parse_error`, `network_error` and `error` and applications simply don't listen to events they are not interested in?